google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemTest.java

```diff
@@ -334,6 +337,43 @@ public void testDeleteRecursive() throws IOException {
     }
   }
 
+  @Test
+  public void testSameProvider() {
+    try (CloudStorageFileSystem sourceFileSystem =
+        CloudStorageFileSystem.forBucket(
+            "bucket",
+            CloudStorageConfiguration.builder().permitEmptyPathComponents(true).build())) {
+      CloudStorageFileSystem destFileSystem =
+          CloudStorageFileSystem.forBucket(
+              "new-bucket",
+              CloudStorageConfiguration.builder().permitEmptyPathComponents(true).build());
+      assertEquals(sourceFileSystem.provider(), destFileSystem.provider());
+      assertEquals(sourceFileSystem.config(), destFileSystem.config());
+      assertEquals("bucket", sourceFileSystem.bucket());
+      assertEquals("new-bucket", destFileSystem.bucket());
+    } catch (IOException e) {
+
+    }
+  }
+
+  @Test
+  public void testDifferentProvider() {
+    try (CloudStorageFileSystem sourceFileSystem =
+        CloudStorageFileSystem.forBucket(
+            "bucket",
+            CloudStorageConfiguration.builder().permitEmptyPathComponents(true).build())) {
+      CloudStorageFileSystem destFileSystem =
+          CloudStorageFileSystem.forBucket(
+              "new-bucket",
+              CloudStorageConfiguration.builder().permitEmptyPathComponents(false).build());
+      assertFalse(sourceFileSystem.provider() == destFileSystem.provider());
+      assertNotEquals(sourceFileSystem.config(), destFileSystem.config());
+      assertEquals("bucket", sourceFileSystem.bucket());
+      assertEquals("new-bucket", destFileSystem.bucket());
+    } catch (IOException e) {
```

Comment:
IOException should fail test; i.e. don't catch it

Reply:
Done